modules/tensor_mechanics/src/userobjects/CavityPressureUserObject.C

```diff
@@ -65,6 +81,24 @@ CavityPressureUserObject::CavityPressureUserObject(const InputParameters & param
   auto volume_names = params.get<std::vector<PostprocessorName>>("volume");
   for (unsigned int i = 0; i < volume_names.size(); ++i)
     _volume[i] = &getPostprocessorValueByName(volume_names[i]);
+
+  if (isParamValid("additional_volumes") && !isParamValid("temperature_of_additional_volumes"))
+    mooseError("If additional volumes are specified, the temperatures of the volumes must also be "
+               "specified.");
+
+  if (!isParamValid("additional_volumes") && isParamValid("temperature_of_additional_volumes"))
+    mooseError(
+        "Additional volumes must be specified when the temperatures of the volumes are specified.");
```

Comment:
You could have the check be (isPV(volume) != isPV(temp)) mooseError(...);  That would check both cases at once.  Just a comment.  Change if you like.

Reply:
I like that change.